src/distribute.ts

```diff
@@ -0,0 +1,135 @@
+import { writeFileSync } from 'fs';
+import { arweave, readInitialOutputFile } from './common';
+import type { Rewards } from './inferno_types';
+import Transaction from 'arweave/node/lib/transaction';
+
+export type TransactionToDistribute = {
+	id: string;
+	tx: Transaction;
+	target: string;
+	qty: number;
+	rankPosition: number;
+};
+
+// Get the key file
+const keyfile = process.env.KEYFILE ? JSON.parse(process.env.KEYFILE) : undefined;
+
+export async function distributeTokens(confirm: boolean) {
+	const data = readInitialOutputFile();
+	const weeklyRank = data.ranks.weekly;
+
+	if (!keyfile) {
+		console.log("There's no keyfile to create transactions");
+		return;
+	}
+
+	if (!weeklyRank.hasReachedMinimumGroupEffort) {
+		console.log("The group didn't reach the minimum effort this week");
+		return;
+	}
+
+	if (confirm) {
+		console.log('========= CREATING TRANSACTIONS =========');
+	} else {
+		console.log('DRY-RUN: THE SCRIPT IS NOT POSTING TRANSACTIONS');
+	}
+
+	const addresses = weeklyRank.groupEffortRewards.filter((wallet) => wallet.rewards > 0);
+	const transactions = await createTransactions(addresses);
+
+	if (confirm) {
+		console.log('========= SENDING TRANSACTIONS =========');
+	}
+
+	const transactionsToReport: { id: string; qty: number; rankPosition: number }[] = [];
+
+	for (const transaction of transactions) {
+		if (!confirm) {
+			console.log(transaction);
+			return;
+		}
+
+		const transactionSent = await sendTransaction(transaction);
+		if (transactionSent) {
+			writeFileSync(`./distribution/${transaction.id}`, JSON.stringify(transaction, null, '\t'));
```

Comment:
The path is perfectly functional for *nix, but it will fail on windows. We use `join` of `fs` in order to concatenate paths 🤓. Non-blocking comment, you are good

Reply:
I used normalize :D
https://github.com/ardriveapp/inferno-backend/pull/24/commits/aa763e8ab8074f5372d22f5e41507a89674ffd13